anchore_engine/services/policy_engine/engine/vulns/cache_managers.py

```diff
@@ -150,7 +147,11 @@ def _get_current_grype_metadata():
         TODO Invoke the grype_wrapper and generate a grype metadata object, work with dspalmer
         """
 
-        return GrypeMetadata("foo", "bar", "sha256:blah")
+        # return GrypeMetadata("", "", "")
+        archive_checksum = (
+            GrypeWrapperSingleton.get_instance().get_current_grype_db_checksum()
+        )
+        return {"checksum": archive_checksum}
```

Comment:
Will this ever be called before the grype wrapper has initialized the db? Im guessing not but just wanted to double check because in my code we recently had to handle a ValueError that gets thrown if the grype_db_dir has not yet been initialized.

Reply:
I noticed. When correctly implemented this won't use the wrapper. It'll instead use the active db record to decide whether the cache is valid or not. That's one of the reasons why the grype-db checksum (not just the archive) must be stored